module/advancement/advancement-config.mjs

```diff
@@ -101,4 +114,86 @@ export default class AdvancementConfig extends FormApplication {
     }, {});
   }
 
+  /* -------------------------------------------- */
+  /*  Drag & Drop for Item Pools                  */
+  /* -------------------------------------------- */
+
+  /**
+   * Handle deleting an existing Item entry from the Advancement.
+   * @param {Event} event        The originating click event.
+   * @returns {Promise<Item5e>}  The updated parent Item after the application re-renders.
+   * @protected
+   */
+  async _onItemDelete(event) {
+    event.preventDefault();
+    const uuidToDelete = event.currentTarget.closest("[data-item-uuid]")?.dataset.itemUuid;
+    if ( !uuidToDelete ) return;
+    const items = foundry.utils.getProperty(this.advancement.data.configuration, this.options.dropKeyPath);
+    const updates = { configuration: await this.prepareConfigurationUpdate({
+      [this.options.dropKeyPath]: items.filter(uuid => uuid !== uuidToDelete)
+    }) };
+    console.log(updates);
+    await this.advancement.update(updates);
+    this.render();
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritdoc */
+  _canDragDrop() {
+    return this.isEditable;
```

Comment:
Idle thought: this should probably be the default behavior of the base `FormApplication`